internal/routing/handlers/user_handler.go

```diff
@@ -437,3 +545,28 @@ func generateToken() string {
 	// Generate a random 6-digit number
 	return strconv.Itoa(rand.Intn(900000) + 100000)
 }
+
+func checkPassword(w http.ResponseWriter, transactionCtx context.Context, tx pgx.Tx, username string, givenPassword string) error {
+	queryString := "SELECT password, user_id FROM alpha_schema.users WHERE username = $1"
```

Comment:
Da du die `user_id` in der Methode nicht verwendest, kannst du das aus der Query hier streichen.